### ISSUE TITLE
Bug - Form(FormGroup) - VoiceOver is not reading mandatory checkboxes and radio buttons as mandatory

### ISSUE THREAD

**Issue by AyushiMidha** [patternfly/patternfly-react#12110]
**Describe the problem**
VoiceOver is not correctly announcing mandatory checkboxes and radio buttons as required within the FormGroup component even when isRequired prop is used.

**How do you reproduce the problem?**

- Navigate to a page using the Forms > Form component that includes required checkboxes or radio buttons(Add isRequired to FormGroup).
- Enable VoiceOver on macOS.
- Move focus to the group of checkbox or radio buttons including their label.
- Observe that VoiceOver does not announce them as "required" or "mandatory."

**Expected behavior**
VoiceOver should announce mandatory checkboxes and radio buttons as “required” when they are marked as such, ensuring that screen reader users are aware of which fields must be filled.

**Screenshots**

<img width="1240" height="332" alt="Image" src="https://github.com/user-attachments/assets/accc1b28-88d7-4455-9e3b-1d9bf1d6116a" />

**What is your environment?**
 - OS:  iOS 5.7.1
 - Browser: Chrome Version 141.0.7390.54 (Official Build)
